### PR TITLE
[google-apps-script] Add driveFile, driveFolder to DriveItem/DriveItemReference

### PR DESCRIPTION
**Pull request by mtgto**
Add driveFile, driveFolder to DriveItem/DriveItemReference, which is alternative obsoleted file and folder properties.

closes https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/66806

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests). It fails with `Exceeds maximum line length of 200`

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/drive/activity/v2/reference/rest/v2/activity/driveitem https://developers.google.com/drive/activity/v2/reference/rest/v2/activity/driveitemreference
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/google-apps-script/test/google-apps-script-tests.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@
 // from https://developers.google.com/apps-script/overview
```
Comment (mtgto) [inline types/google-apps-script/test/google-apps-script-tests.ts:3]: prettier converts quote to singlequote.

File: types/google-apps-script/apis/driveactivity_v2.d.ts [hunk 2]
```diff
@@ -77,18 +77,26 @@ declare namespace GoogleAppsScript {
                 timestamp?: string | undefined;
             }
```
Comment (mtgto) [inline types/google-apps-script/apis/driveactivity_v2.d.ts:78]: https://developers.google.com/drive/activity/v2/reference/rest/v2/activity/driveitem

File: types/google-apps-script/apis/driveactivity_v2.d.ts [hunk 3]
```diff
@@ -77,18 +77,26 @@ declare namespace GoogleAppsScript {
...
                 file?: any;
+                // folder is deprecated; please use the driveFolder instead.
                 folder?: DriveActivity.Schema.Folder | undefined;
                 mimeType?: string | undefined;
                 name?: string | undefined;
                 owner?: DriveActivity.Schema.Owner | undefined;
                 title?: string | undefined;
+                driveFile?: any;
+                driveFolder?: DriveActivity.Schema.Folder;
             }
```
Comment (mtgto) [inline types/google-apps-script/apis/driveactivity_v2.d.ts:90]: https://developers.google.com/drive/activity/v2/reference/rest/v2/activity/driveitemreference

File: types/google-apps-script/apis/driveactivity_v2.d.ts [hunk 4]
```diff
@@ -77,18 +77,26 @@ declare namespace GoogleAppsScript {
                 timestamp?: string | undefined;
             }
             interface DriveItem {
+                // file is deprecated; please use the driveFile instead.
```
Comment (azatoth) [inline types/google-apps-script/apis/driveactivity_v2.d.ts:80]: perhaps use jsdoc `@deprecated`?

File: types/google-apps-script/apis/driveactivity_v2.d.ts [hunk 5]
```diff
@@ -77,18 +77,26 @@ declare namespace GoogleAppsScript {
...
                 file?: any;
+                // folder is deprecated; please use the driveFolder instead.
                 folder?: DriveActivity.Schema.Folder | undefined;
                 mimeType?: string | undefined;
                 name?: string | undefined;
                 owner?: DriveActivity.Schema.Owner | undefined;
                 title?: string | undefined;
+                driveFile?: any;
```
Comment (azatoth) [inline types/google-apps-script/apis/driveactivity_v2.d.ts:88]: Should be an object and not any; perhaps also add a File to Schema?

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/google-apps-script/test/google-apps-script-tests.ts
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 04fe91de32e637df4c867310191f69b31146f843
</p>

File: types/google-apps-script/apis/driveactivity_v2.d.ts (hunk 4 above)
Comment (jakebailey) [reply types/google-apps-script/apis/driveactivity_v2.d.ts:80]:
```suggestion
                /** @deprecated Use driveFile instead. */
```

File: types/google-apps-script/apis/driveactivity_v2.d.ts [hunk 6]
```diff
@@ -77,18 +77,26 @@ declare namespace GoogleAppsScript {
                 timestamp?: string | undefined;
             }
             interface DriveItem {
+                // file is deprecated; please use the driveFile instead.
                 file?: any;
+                // folder is deprecated; please use the driveFolder instead.
```
Comment (jakebailey) [inline types/google-apps-script/apis/driveactivity_v2.d.ts:82]:
```suggestion
                /** @deprecated Use driveFolder instead. */
```

File: types/google-apps-script/apis/driveactivity_v2.d.ts (hunk 5 above)
Comment (jakebailey) [reply types/google-apps-script/apis/driveactivity_v2.d.ts:88]: The existing type for `file` here is `any`, so I'm not sure that it actually is a downgrade, but I personally agree that a stronger type would be good.

**Review by jakebailey — COMMENTED** [review]
Just left some suggestion comments to apply for the deprecations, but otherwise looks good to me. Will apply the suggestions and approve.

**Review by jakebailey — APPROVED** [review]

**Comment by mtgto** [comment]
@jakebailey Thank you for your review.
I'm a maintainer of types/google-apps-script, forgotting pull request author can not approve 😅

**Comment by mtgto** [comment]
Ready to merge

**Pull request merged**